crates/nu-parser/src/parser.rs

```diff
@@ -855,72 +872,88 @@ pub fn parse_internal_call(
-                if let Some(arg_shape) = flag.arg {
-                    if let Some(arg) = spans.get(spans_idx + 1) {
-                        let (arg, err) =
-                            parse_value(working_set, *arg, &arg_shape, expand_aliases_denylist);
-                        error = error.or(err);
 
-                        if flag.long.is_empty() {
-                            if let Some(short) = flag.short {
+            if matches!(err, Some(ParseError::UnknownFlag(_, _, _, _)))
+                && signature.allows_unknown_args
+            {
+                let (arg, arg_err) = parse_value(
+                    working_set,
+                    arg_span,
+                    &SyntaxShape::Any,
+                    expand_aliases_denylist,
+                );
+
+                call.add_unknown(arg);
+                error = error.or(arg_err);
+            } else {
+                error = error.or(err);
+                for flag in short_flags {
+                    if let Some(arg_shape) = flag.arg {
+                        if let Some(arg) = spans.get(spans_idx + 1) {
+                            let (arg, err) =
+                                parse_value(working_set, *arg, &arg_shape, expand_aliases_denylist);
+                            error = error.or(err);
+
+                            if flag.long.is_empty() {
+                                if let Some(short) = flag.short {
+                                    call.add_named((
+                                        Spanned {
+                                            item: String::new(),
+                                            span: spans[spans_idx],
+                                        },
+                                        Some(Spanned {
+                                            item: short.to_string(),
+                                            span: spans[spans_idx],
+                                        }),
+                                        Some(arg),
+                                    ));
+                                }
+                            } else {
                                 call.add_named((
                                     Spanned {
-                                        item: String::new(),
+                                        item: flag.long.clone(),
                                         span: spans[spans_idx],
                                     },
-                                    Some(Spanned {
-                                        item: short.to_string(),
-                                        span: spans[spans_idx],
-                                    }),
+                                    None,
                                     Some(arg),
                                 ));
                             }
+                            spans_idx += 1;
                         } else {
+                            error = error.or_else(|| {
```

Comment:
Interesting, yeah `git push --force-with-lease` doesn't work. I think we should still report missing flag parameter if the flag requires it. `extern spam [--foo: string]` should report error if you call just `spam --foo`. But `git push --force-with-lease` doesn't require a parameter, so there is something wrong here.

Reply:
Actually #5103 is also a problem of how the completion is defined, like #6020. It is currently not possible for Nushell call parsing to support both `--flag` and `--flag value` at the same time because it is not clear whether `value` is a flag's argument or a positional.